index.bs

```diff
@@ -1288,100 +1344,133 @@ partial interface ModelBuilder {
 
-    **Returns:** an {{Operand}}. The permuted or transposed N-D tensor. 
+    **Returns:** an {{MLOperand}}. The permuted or transposed N-D tensor. 
 </div>
 
-## Model ## {#api-model}
-The {{Model}} interface represents an immutable computational graph constructed by the {{ModelBuilder}} interface. The operands to the starting operations of the graph are assumed to be the model's input operands.
+## MLModel ## {#api-mlmodel}
+The {{MLModel}} interface represents an immutable computational graph constructed by the {{MLModelBuilder}} interface. The operands to the starting operations of the graph are assumed to be the model's input operands.
 
 <script type=idl>
-enum PowerPreference {
-  // Let the user agent decide the most suitable behavior. This is the default value.
-  "default",
-  // Prioritizes execution speed over other considerations e.g. power consumption
-  "high-performance",
-  // Prioritizes power consumption over other considerations e.g. execution speed
-  "low-power"
+[SecureContext, Exposed=Window]
+interface MLModel {
+  Promise<MLCompilation> compile();
 };
+</script>
+
+## Compilation ## {#api-mlcompilation}
+The {{MLCompilation}} interface represents a compiled version of the computational graph of a {{MLModel}} according to the compilation options. A compilation of a graph once constructed is immutable and cannot be subsequently changed.
 
-dictionary CompilationOptions {
-  // Compilation preference as related to power consumption level
-  PowerPreference powerPreference = "default";
+<script type=idl>
+// WebGL specification:
+// (https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.9)
+[SecureContext, Exposed=Window]
+interface WebGLTexture {
+  // ...
 };
 
+// WebGPU specification:
+// (https://gpuweb.github.io/gpuweb/#texture-interface)
 [SecureContext, Exposed=Window]
-interface Model {
-  Promise<Compilation> compile(optional CompilationOptions options = {});
+interface GPUTexture {
+  // ...
 };
-</script>
 
-## Compilation ## {#api-compilation}
-The {{Compilation}} interface represents a compiled version of the computational graph of a {{Model}} according to the compilation options. A compilation of a graph once constructed is immutable and cannot be subsequently changed.
+dictionary MLInput {
+  required ArrayBufferView bufferView;
+  sequence<long> dimensions;
+};
 
-<script type=idl>
-dictionary Input {
-  required ArrayBufferView buffer;
+dictionary MLOutput {
+  ArrayBufferView bufferView;
   sequence<long> dimensions;
 };
 
-dictionary Output {
-  ArrayBufferView buffer;
+dictionary MLWebGLInput {
+  required WebGLTexture glTexture;
   sequence<long> dimensions;
 };
 
-typedef record<DOMString, Input> NamedInputs;
-typedef record<DOMString, Output> NamedOutputs;
+dictionary MLWebGLOutput {
+  WebGLTexture glTexture;
+  sequence<long> dimensions;
+};
+
+dictionary MLGPUInput {
+  required GPUTexture gpuTexture;
```

Comment:
Should we consider to support `GPUBuffer` as input? Given TensorFlow.js WebGPU backend uses GPUBuffer as input and output: https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/backend_webgpu.ts#L42. @pyu10055 for comments.

Reply:
I think it's a good idea. Thanks @huningxin

Reply:
+1 WebGPU will be replacing WebGL, providing support for WebGPU from the start will future-proof the API.

Reply:
Fixed.

Reply:
thanks!